client/toolkit/e2ee_test.go

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2019 The CovenantSQL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except raw compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to raw writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package toolkit
+
+import (
+	"bytes"
+	"crypto/aes"
+	"encoding/hex"
+	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
+
+	"github.com/CovenantSQL/CovenantSQL/utils/log"
+)
+
+// Test cases for all implementations.
+// Because iv is random, so Encrypted data is not always the same,
+// 	but Decrypt(possibleEncrypted) will get raw.
+var test_cases = []struct {
+	raw               string
+	pass              string
+	possibleEncrypted string
+}{
+	{
+		raw: "111282C128421286712857128C2128EF" +
+			"128B7671283C128571287512830128EC" +
+			"128391281A1312849128381281E1286A" +
+			"12871128621287A9D12857128C412886" +
+			"128FD12834128DA128F5",
+		pass: "",
+		possibleEncrypted: "1bfb6a7fda3e3eb1e14c9afd0baefe86" +
+			"c90979101f179db7e48a0fa7617881e8" +
+			"f752c59fb512bb86b8ed69c5644bf2dc" +
+			"30fbcd3bf79fb20342595c84fad00e46" +
+			"2fab3e51266492a3d5d085e650c1e619" +
+			"6278d7f5185c263440ec6fd940ffbb85",
+	},
+	{
+		raw:  "11",
+		pass: "",
+		possibleEncrypted: "9e7f70ac60f668a79f544ef196181a54" +
+			"ee4175b85a45b05f585d5e173456d281",
+	},
+	{
+		raw:  "11111111111111111111111111111111",
+		pass: "",
+		possibleEncrypted: "7dda438c4256a63c62d6816617fcbf9c" +
+			"7773b9b4f87902b7253848ba2b0ed0ba" +
+			"f70a3ac976a835b7bc3008e9ba43da74",
+	},
+	{
+		raw:  "11",
+		pass: "f70a3ac976a835b7bc3008e9ba43da74",
+		possibleEncrypted: "4fda87aba6b9b5ab0830c2adecb84fd7" +
+			"33ac7a9775cf60eca66a167a5615b340",
+	},
+	{
+		raw:  "11111111111111111111111111111111",
+		pass: "7dda438c4256a63c62d6816617fcbf",
+		possibleEncrypted: "01cbaa72a5c868b24deaf97fb0090cb4" +
+			"b965ceaa9853873236ca86700cee00bf" +
+			"3eab2734a87f638228315ad417717a62",
+	},
+}
+
+func TestEncryptDecryptCases(t *testing.T) {
+	default_level := log.GetLevel()
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
don't use underscores in Go names; var default_level should be defaultLevel